test/functional/p2p_sendrecon.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test SENDRECON message
+"""
+
+import time
+
+from test_framework.messages import (
+    msg_sendrecon,
+    msg_verack,
+    msg_version,
+)
+from test_framework.p2p import (
+    P2PInterface,
+    P2P_SERVICES,
+    P2P_SUBVERSION,
+)
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+class SendReconSender(P2PInterface):
+    def __init__(self, manual_verack=False):
+        super().__init__()
+        self.version_received = False
+        self.sendrecon_msg_received = None
+        self.manual_verack = manual_verack
+
+    def on_version(self, message):
+        if not self.manual_verack:
+            super().on_version(message)
+        self.version_received = True
+
+    def on_sendrecon(self, message):
+        print(message)
```

Comment:
in d3cc218415f3a5f9a7170cda48a5e3938900832c:

Remove this print? Or at least use `self.log`